File: xchange-idex/src/main/java/org/knowm/xchange/idex/IdexWithdraw.java
package org.knowm.xchange.idex;

import java.math.BigInteger;
import org.knowm.xchange.idex.dto.WithdrawReq;
import org.knowm.xchange.service.trade.params.WithdrawFundsParams;

final class IdexWithdraw extends WithdrawReq implements WithdrawFundsParams {
  public IdexWithdraw(
      String address,
      String amount,
      String token,
      BigInteger nonce,
      String s,
      BigInteger v,
      String r) {
    address(address).amount(amount).token(token).nonce(nonce).s(s).v(v).r(r);
  }
}

File: xchange-idex/src/main/java/org/knowm/xchange/idex/IdexWithdraw.kt
package org.knowm.xchange.idex


import org.knowm.xchange.idex.dto.*
import org.knowm.xchange.service.trade.params.*
import java.math.*

class IdexWithdraw(address: String, amount: String, token: String, nonce: BigInteger, s: String, v: BigInteger,
                   r: String) : WithdrawReq(), WithdrawFundsParams {init {
    address(address)
            .amount(amount)
            .token(token)
            .nonce(nonce)
            .s(s)
            .v(v)
            .r(r)
}
}

